reportssql/report_util_functions.sql

```diff
@@ -1483,4 +1483,223 @@ BEGIN
+    DECLARE patientHIVRetestResultIsPositive TINYINT(1) DEFAULT 0;
+    DECLARE patientRepeatTestDateIsBetweenReportingPeriod TINYINT(1) DEFAULT 0;
+    DECLARE uuidRepeatTestDone VARCHAR(38) DEFAULT "4026c752-ecfe-43b0-9cf9-14a74d961fd1";
+    DECLARE uuidRepeatTestIfYes VARCHAR(38) DEFAULT "a2065636-5326-40f5-aed6-0cc2cca81ccc";
+    DECLARE uuidRepeatTestResult VARCHAR(38) DEFAULT "7682c09b-8e81-4e30-8afd-636fb9fcd4a1";
+    DECLARE uuidRepeatTestResultIsPositive VARCHAR(38) DEFAULT "7acfafa4-f19b-485e-97a7-c9e002dbe37a";
+    DECLARE uuidRepeatTestDate VARCHAR(38) DEFAULT "541d9f7b-f622-4ebc-a3a3-50c970d4cce0";
+
+    SELECT
+        TRUE INTO patientRetestedForHIVWIthinReportingPeriod
+    FROM obs o
+    JOIN concept c ON c.concept_id = o.concept_id AND c.retired = 0
+    WHERE o.voided = 0
+        AND o.person_id = p_patientId
+        AND c.uuid = uuidRepeatTestDone
+        AND o.value_coded IS NOT NULL
+        AND o.value_coded = (SELECT concept_id FROM concept WHERE uuid = uuidRepeatTestIfYes)
+        AND DATE(o.obs_datetime) BETWEEN p_startDate AND p_endDate
+        LIMIT 1;
+    
+    SELECT
+        TRUE INTO patientHIVRetestResultIsPositive
+    FROM obs o
+    JOIN concept c ON c.concept_id = o.concept_id AND c.retired = 0
+    WHERE o.voided = 0
+        AND o.person_id = p_patientId
+        AND c.uuid = uuidRepeatTestResult
+        AND o.value_coded IS NOT NULL
+        AND o.value_coded = (SELECT concept_id FROM concept WHERE uuid = uuidRepeatTestResultIsPositive)
+        LIMIT 1;
+
+    SELECT
+        TRUE INTO patientRepeatTestDateIsBetweenReportingPeriod
+    FROM obs o
+    JOIN concept c ON c.concept_id = o.concept_id AND c.retired = 0
+    WHERE o.voided = 0
+        AND o.person_id = p_patientId
+        AND c.uuid = uuidRepeatTestDate
+        AND DATE(o.value_datetime) BETWEEN p_startDate AND p_endDate
+        LIMIT 1;
+
+    RETURN (patientRetestedForHIVWIthinReportingPeriod && patientHIVRetestResultIsPositive && patientRepeatTestDateIsBetweenReportingPeriod);
+END$$
+DELIMITER ;
+
+-- hivStatusKnownMoreThan3MonthsBeforeReportEndDate
+
+DROP FUNCTION IF EXISTS hivStatusKnownMoreThan3MonthsBeforeReportEndDate;
+
+DELIMITER $$
+CREATE FUNCTION hivStatusKnownMoreThan3MonthsBeforeReportEndDate(
+    p_patientId INT(11),
+    p_endDate DATE) RETURNS TINYINT(1)
+    DETERMINISTIC
+BEGIN
+    DECLARE result TINYINT(1) DEFAULT 0;
+    DECLARE uuidHIVTestDate VARCHAR(38) DEFAULT "c6c08cdc-18dc-4f42-809c-959621bc9a6c";
+
+    SELECT
+        TRUE INTO result
+    FROM obs o
+    JOIN concept c ON c.concept_id = o.concept_id AND c.retired = 0
+    WHERE o.voided = 0
+        AND o.person_id = p_patientId
+        AND c.uuid = uuidHIVTestDate
+        AND o.value_datetime IS NOT NULL
+        AND timestampdiff(MONTH, CAST(o.value_datetime AS DATE), p_endDate) > 3
+        LIMIT 1;
+
+    RETURN (result );
+END$$
+DELIMITER ;
+
+-- patientIsFemale
```

Comment:
@prashantsable1991 Please delete this function and use patientGenderIs(patient_id, 'F') instead